src/core/server/rendering/bootstrap/bootstrap_renderer.ts

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { createHash } from 'crypto';
+import * as UiSharedDeps from '@kbn/ui-shared-deps';
+import { PackageInfo } from '@kbn/config';
+import { UiPlugins } from '../../plugins';
+import { IUiSettingsClient } from '../../ui_settings';
+import { HttpAuth, KibanaRequest } from '../../http';
+import { getPluginsBundlePaths } from './get_plugin_bundle_paths';
+import { renderTemplate } from './render_template';
+
+export type BootstrapRendererFactory = (factoryOptions: FactoryOptions) => BootstrapRenderer;
+export type BootstrapRenderer = (options: RenderedOptions) => Promise<RendererResult>;
+
+interface FactoryOptions {
+  serverBasePath: string;
+  packageInfo: PackageInfo;
+  uiPlugins: UiPlugins;
+  auth: HttpAuth;
+}
+
+interface RenderedOptions {
+  request: KibanaRequest;
+  uiSettingsClient: IUiSettingsClient;
+}
+
+interface RendererResult {
+  body: string;
+  etag: string;
+}
+
+export const bootstrapRendererFactory: BootstrapRendererFactory = ({
```

Comment:
optional: this file contains some logic, but I don't see `bootstrapRendererFactory` is tested